tests/workloads/ai_dynamo/test_command_gen_strategy_slurm.py

```
@@ -218,6 +220,169 @@ def test_gen_script_args_contains_custom_aiperf_accuracy_args(strategy: AIDynamo
+
+def test_generated_aiperf_script_rejects_list_args(strategy: AIDynamoSlurmCommandGenStrategy) -> None:
+    td = cast(AIDynamoTestDefinition, strategy.test_run.test)
+    td.cmd_args.workloads = "aiperf.sh"
+    td.cmd_args.aiperf = AIPerf.model_validate({"args": {"server-metrics-formats": ["json", "csv"]}})
+
+    with pytest.raises(ValueError, match="AIPerf argument 'server-metrics-formats' must be a scalar value"):
+        strategy._gen_script_args(td)
+
+
+def test_aiperf_extra_args_must_be_string() -> None:
+    with pytest.raises(ValueError):
```

Ruff (0.15.15)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)